### PR TITLE
My Home: Add Promote Post to my home page swipeable

### PR DESCRIPTION
**Pull request by benzhu56**
#### Intro
Hi Team! This is my first PR in Calypso and the first PR of a series of PRs to add entry points in Calypso for a new feature: allow WordPress.com users to pay for advertising to promote their posts on various A8C products. 
For details please see this P2 A self-serve ad client for WPCOM (pbpazC-2Pd-p2).  
The majority of the feature is done outside of WP.com as a standalone widget, here we only need to add buttons to Calypso to show the widget with required parameters.
This PR does **NOT** have all the code for the feature to work. In order to make the PRs smaller, we've put it behind a **feature flag `promote-post`**, and it's hidden for now. So we can iterate quickly.

#### Proposed Changes
This PR adds a swipeable task to My Home page on top, when the promote-post feature is enabled, it will show up and allow user to click on the button to invoke the widget (or jump to another route, the behaviour is not set in stone yet). 

#### Testing Instructions
Since the feature is behind a toggle and hidden. Also the actual widget isn't deployed yet. So here we only need to test that, when the feature is not enabled, it does not show up in the task list. And when it is enabled, it should show up. (No need to test clicking the button, we'll iterate on that in later PRs).

1. Check out the branch.
2. Load up the My Home page of your favourite testing blog, check that the Promote Post is NOT there.
3. Enable the `promote-post` feature, and load the page again. Check now that the Promote Post should show up.

Screen Shot 2022-07-27 at 10 12 07

*

#### Pre-merge Checklist

- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [x] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have you sent any new strings for translation (PCYsg-1vr-p2) ASAP? (We'll do that once the strings are final, I'll create a ticket for that).

### CODE REVIEW

File: client/lib/dsp/index.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+		}
+		return;
+	}
+	const script = document.createElement( 'script' );
+	script.src = config( 'dsp_widget_js_src' );
+	script.async = true;
+	if ( onLoad ) {
+		script.onload = onLoad;
+	}
+	document.body.appendChild( script );
```
Comment (benzhu56) [inline client/lib/dsp/index.ts]: The feature requires us to load a bootstrap js from WP.com as the widget is a standalone one. 
Is this the correct way of injecting the js?

**Review by benzhu56 — COMMENTED** [review]
1. Haven't added any tests yet, the feature is iterating quickly, is it ok to add tests later when things aren't changing quickly?
2. I can't seem to test this in jurassic.ninja because it's not using Calypso, is that right?

**Comment by benzhu56** [comment]
FYI: @jjolmo @jamesgill3 
I've created this PR and let's see if we can just merge quickly into Calypso instead of stacking the PRs. This way it will make our work easier at a later stage.

File: client/lib/dsp/index.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+		await loadDSPWidgetJS( async () => await showDSPWidgetModal( siteId, postId ) );
+	} else {
+		await window.BlazePress.render( {
+			stripeKey: config( 'dsp_stripe_pub_key' ),
+			apiHost: 'https://public-api.wordpress.com',
+			apiPrefix: `/wpcom/v2/sites/${ siteId }/wordads/dsp`,
+			// todo fetch rlt somehow
+			authToken: 'wpcom-proxy-request',
+			template: 'article',
+			urn: `urn:wpcom:post:${ siteId }:${ postId || 0 }`,
```
Comment (j6ll) [inline client/lib/dsp/index.ts]: Hey @benz1924 I'm not sure we should be sending 0 here?  Maybe this should be null?

We need to fix this in the dsp project, but I'm getting errors when I progress in the widget becuase the post id = 0, but a site id is present.

Screenshot 2022-07-27 at 18 52 52

Comment (j6ll) [reply client/lib/dsp/index.ts]: Created a ticket for this (No. 346) in our project board - don't mind who picks this up.  I'm also happy to take a look!

Comment (benzhu56) [reply client/lib/dsp/index.ts]: Yeah, I wasn't sure about the action of the button either. I put value 0 just wanted to see the modal show up. 
There were some discussions around whether clicking the button should open the modal or should go to a landing page of in Calypso.
Before we get a clear answer, I guess we can leave it there. It's behind a feature toggle, so should be safe.

**Comment by benzhu56** [comment]
@Automattic/team-calypso 
Hi team Calypso, first time contributor here, could you please help review the PR, especially following parts:
1. This is a starting PR of a series of PRs, hidden behind a feature flag, WIP, but we would like to merge so multiple developers can iterate on this. (Also keeping PRs small. :)
2. As listed in the comment https://github.com/Automattic/wp-calypso/pull/65996#discussion_r930456349 , we'll need to load a standalone widget and please help verify if we are using the desired way of doing that. 
Thanks!

File: client/lib/dsp/README.md [hunk 3]
```diff
@@ -0,0 +1,3 @@
+# DSP Widget
+
+
```
Comment (bluefuton) [inline client/lib/dsp/README.md]: Sorry if this is a silly question, but what does DSP stand for? Can we use a more descriptive name?

Comment (benzhu56) [reply client/lib/dsp/README.md]: I was being silly! I've updated it with a bit more info, but it's likely we'll update it again along the way. Thanks for looking at this @bluefuton , appreciate it!

**Review by gravityrail — APPROVED** [review]
Looks great

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7461157

Thank you @benzhu56 for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7461157).

**Comment by wongasy** [comment]
Hi @benzhu56 :) I was testing a `Swipeable` fix in [another PR](https://github.com/Automattic/wp-calypso/pull/67162) and noticed an issue related to the Promote Post slide. Thought I'd flag it to you. In the case where `showPromotePost` is `false,` we end up with a blank slide:
https://github.com/Automattic/wp-calypso/blob/dc9fe78154ce60eb84aae1da88ef68b03f1bf9fe/client/my-sites/customer-home/cards/tasks/promote-post/index.js#L17

Screen Shot 2022-08-31 at 8 41 24 PM

Screen Shot 2022-08-31 at 8 39 56 PM

**Comment by benzhu56** [comment]
Ah, thanks for spotting it @wongasy ! We'll get it fixed.

**Pull request merged**